instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/AwsXrayEnvCarrierEnricher.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awslambdacore.v1_0.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Enrich carrier with AWS Lambda runtime trace information.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class AwsXrayEnvCarrierEnricher {
+
+  private static final String AWS_TRACE_ENV_VAR = "_X_AMZN_TRACE_ID";
+  private static final String AWS_TRACE_PROPERTY = "com.amazonaws.xray.traceHeader";
+  private static final String AWS_TRACE_PROPAGATOR_KEY = "x-amzn-trace-id";
```

Comment:
my preference would be for this to be a separate key in the carrier, e.g. `x-amzn-env-trace-id` or `x-amzn-vended-trace-id` (instead of overwriting the AwsLambdaRequest header), so that the propagator chain can have access to all of the different propagated trace contexts

but this is a question that needs to be answered on the spec side